src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -51,16 +55,159 @@ graph_status_to_string(int status) {
   }
 }
 
+void
+populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& tile, int row, int col)
+{
+  boost::property_tree::ptree pt;
+  std::string aie_data;
+  std::vector<std::string> graph_status;
+  boost::property_tree::ptree tile_array;
+  boost::property_tree::ptree empty_pt;

```

Comment:
This looks suspiciously like C style coding.  Please move variable declarations/definitions to where they are first used, or as close as possible.